homeassistant/components/swiss_public_transport/coordinator.py

```diff
@@ -58,41 +60,51 @@ async def _async_update_data(self) -> DataConnection:
             )
             raise UpdateFailed from e
 
-        departure_time = (
-            dt_util.parse_datetime(self._opendata.connections[0]["departure"])
-            if self._opendata.connections[0] is not None
-            else None
-        )
-        next_departure_time = (
-            dt_util.parse_datetime(self._opendata.connections[1]["departure"])
-            if self._opendata.connections[1] is not None
-            else None
-        )
-        next_on_departure_time = (
-            dt_util.parse_datetime(self._opendata.connections[2]["departure"])
-            if self._opendata.connections[2] is not None
-            else None
-        )
+        data_connections = []
 
-        if departure_time:
-            remaining_time = departure_time - dt_util.as_local(dt_util.utcnow())
-        else:
-            remaining_time = None
-
-        return DataConnection(
-            departure=departure_time,
-            next_departure=next_departure_time.isoformat()
-            if next_departure_time is not None
-            else None,
-            next_on_departure=next_on_departure_time.isoformat()
-            if next_on_departure_time is not None
-            else None,
-            train_number=self._opendata.connections[0]["number"],
-            platform=self._opendata.connections[0]["platform"],
-            transfers=self._opendata.connections[0]["transfers"],
-            duration=self._opendata.connections[0]["duration"],
-            start=self._opendata.from_name,
-            destination=self._opendata.to_name,
-            remaining_time=f"{remaining_time}",
-            delay=self._opendata.connections[0]["delay"],
-        )
+        for i in range(3):
```

Comment:
I'm not sure why we do this now. We still only display 3 trips right?

Reply:
For now yes, but the api supports a limit param which I want to introduce in the future

Reply:
So you want people to create X amount of sensors? Why don't you just create everything and disable everything above 3 or something like that?

Reply:
It can go up to 15 connections at max. Should I just create all of them and deactive the ones not loaded/available?

Reply:
For now it is fixed at 3, but I want to change that in the future.

Reply:
Hmm, I'm really thinking about the use case for knowing 15 busses ahead. Why would anyone want to know that (I know you want to make it configurable).

Did you think about putting certain data in a service call?

Reply:
Yes, a service call is definitely something I want to add as well. But just because the api-limit is 15 does not mean it makes sense to use it. But I think up to 5 or 6 connections might be useful. For example for a bus connection to have all connections in the same hour with a `12min` cycle could be useful. Therefore I really don't know how many sensors I should create and have to let the user "decide" that. But if you can show be a more dynamic way of setting it up instead of creating as many as the limit specifies and deactivating of no data is available, I am curious.

Reply:
I can imagine also, that a some people will set the limit to `1` quite often.

Reply:
Also note, I'm thinking from an ideology of making a one size fits all approach. When you look at your config entries, the amount of integrations with an options flow is not a lot. (Or at least, the ones that are there aren't usually of the kind that ask the user "how many entities do you want me to show?") 

Less configuring is more.

Reply:
Then how would you let the user control how many connections he needs? I can also just set the limit to 15 and don't care? But I don't really like that.

Reply:
The limit will stay at three, if more is needed, a service call will be implemented in a future PR.